### PR TITLE
Scaling of wider number of resources. 

### PR DESCRIPTION
**Pull request by kenan435**
fixes https://github.com/kubernetes/dashboard/issues/1560

I did this only for deployments, and replication controllers atm. Need some some feedback if its the right direction.... The refactoring will make it easy to add scaling feature to other resources.

### CODE REVIEW

**Comment by maciaszczykm** [comment]
I tested it from user perspective on `gulp serve:prod` and it works. After scaling Deployment number of assigned pod changes. 

***

There is one bug, which probably isn't connected to your pull request:

image

It should say 2/2 after scaling, but it isn't. Can you take a look on it? If it's not your change, then let's create an issue on it and work on it later.

File: src/app/backend/handler/apihandler.go [hunk 1]
```diff
@@ -381,6 +383,16 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 		apiV1Ws.GET("/deployment/{namespace}/{deployment}/oldreplicaset").
 			To(apiHandler.handleGetDeploymentOldReplicaSets).
 			Writes(replicasetlist.ReplicaSetList{}))
+
+	apiV1Ws.Route(
+		apiV1Ws.PUT("/scale/{kind}/{namespace}/{name}/{count}").
+			To(apiHandler.handleUpdateReplicaCount).
+			Writes(scale.ReplicaCounts{}))
+	apiV1Ws.Route(
+		apiV1Ws.GET("/scale/{kind}/{namespace}/{name}").
```
Comment (maciaszczykm) [inline src/app/backend/handler/apihandler.go]: I like, that it's generic and should be able to handle all of scalable resources.

File: src/app/backend/handler/apihandler.go [hunk 2]
```diff
@@ -381,6 +383,16 @@ func CreateHTTPAPIHandler(client *clientK8s.Clientset, heapsterClient client.Hea
 		apiV1Ws.GET("/deployment/{namespace}/{deployment}/oldreplicaset").
 			To(apiHandler.handleGetDeploymentOldReplicaSets).
 			Writes(replicasetlist.ReplicaSetList{}))
+
+	apiV1Ws.Route(
+		apiV1Ws.PUT("/scale/{kind}/{namespace}/{name}/{count}").
```
Comment (maciaszczykm) [inline src/app/backend/handler/apihandler.go]: Hmm... I'm not sure if `{count}` shouldn't be `QueryParameter`.

File: src/app/backend/resource/scaling/scale.go [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+func UpdateReplicaCount(client client.Interface, kind, namespace, name, count string) (rc *ReplicaCounts, err error) {
+	rc = new(ReplicaCounts)
+	s, err := client.Extensions().Scales(namespace).Get(kind, name)
+	if err != nil {
+		return nil, err
+	}
+
+	//convert string to int, not sure if there is an easier way to do this
```
Comment (maciaszczykm) [inline src/app/backend/resource/scaling/scale.go]: It's good, you can remove comment :-)

File: src/app/backend/resource/scaling/scale.go [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	s, err = client.Extensions().Scales(namespace).Update(kind, s)
+	if err != nil {
+		return nil, err
+	}
+
+	rc.DesiredReplicas = s.Spec.Replicas
+	rc.ActualReplicas = s.Status.Replicas
+
+	return
+}
```
Comment (maciaszczykm) [inline src/app/backend/resource/scaling/scale.go]: Add trailing lines at end of each file.

File: src/app/frontend/common/replicas/updatereplicas_dialog.js [hunk 5]
```diff
@@ -12,29 +12,30 @@
...
  * @param {string} namespace
- * @param {string} replicationController
+ * @param {string} resourceName
  * @param {number} currentPods
  * @param {number} desiredPods
  * @return {!angular.$q.Promise}
  */
 export default function showUpdateReplicasDialog(
-    mdDialog, namespace, replicationController, currentPods, desiredPods) {
+    mdDialog, namespace, resourceName, currentPods, desiredPods, resourceKindName) {
```
Comment (maciaszczykm) [inline src/app/frontend/common/replicas/updatereplicas_dialog.js]: `resourceKindName` is not in comment.

**Comment by maciaszczykm** [comment]
image

It seems confusing to me, that current number of replicas is displayed in input field, which cannot be edited. Can you change it?

**Comment by maciaszczykm** [comment]
Updated branch to calculate backend coverage.

**Comment by maciaszczykm** [comment]
Going to take a look on it now.

**Comment by kenan435** [comment]
Its missing some of the tests now... I started to work on that.

**Comment by maciaszczykm** [comment]
@kenan435 Just ping one of use when finished.

**Comment by floreks** [comment]
There are some conflicts. Please resolve them.

**Comment by kenan435** [comment]
Resolved conflicts.

**Pull request closed without merging**